models/User.go

```diff
@@ -1132,6 +1164,7 @@ func saveBidCancelOrderTransaction(bidOrder *Bid, user *User, tx *gorm.DB) error
 	)
 
 	user.Cash += uint64(reservedCash)
+	user.ReservedCash -= uint64(reservedCash)
```

Comment:
`saveBidCancelOrderTransaction` is called by `CancelOrder` in User.go. If an error is thrown by `saveBidCancelOrderTransaction`, a tx.Rollback() is called. However, we are not reverting to the old state.

This was not needed before, because we were not changing the user's cash or reserved cash. But now since we introduced reserving stocks and cash, we need to also revert to old state on top of rolling the database back. 

So CancelOrder() needs to be updated with this.

Reply:
And also add a blank line after this